zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/job/JobUpdateProcessor.java

```diff
@@ -9,19 +9,59 @@
 
 import io.camunda.zeebe.engine.processing.job.behaviour.JobUpdateBehaviour;
 import io.camunda.zeebe.engine.processing.streamprocessor.TypedRecordProcessor;
+import io.camunda.zeebe.engine.processing.streamprocessor.writers.StateWriter;
+import io.camunda.zeebe.engine.processing.streamprocessor.writers.TypedRejectionWriter;
+import io.camunda.zeebe.engine.processing.streamprocessor.writers.TypedResponseWriter;
+import io.camunda.zeebe.engine.processing.streamprocessor.writers.Writers;
 import io.camunda.zeebe.protocol.impl.record.value.job.JobRecord;
+import io.camunda.zeebe.protocol.record.RejectionType;
+import io.camunda.zeebe.protocol.record.intent.JobIntent;
 import io.camunda.zeebe.stream.api.records.TypedRecord;
+import java.util.ArrayList;
+import java.util.List;
 
 public class JobUpdateProcessor implements TypedRecordProcessor<JobRecord> {
 
   private final JobUpdateBehaviour jobUpdateBehaviour;
+  private final TypedRejectionWriter rejectionWriter;
+  private final TypedResponseWriter responseWriter;
+  private final StateWriter stateWriter;
 
-  public JobUpdateProcessor(final JobUpdateBehaviour jobUpdateBehaviour) {
+  public JobUpdateProcessor(final JobUpdateBehaviour jobUpdateBehaviour, final Writers writers) {
     this.jobUpdateBehaviour = jobUpdateBehaviour;
+    rejectionWriter = writers.rejection();
+    responseWriter = writers.response();
+    stateWriter = writers.state();
   }
 
   @Override
   public void processRecord(final TypedRecord<JobRecord> command) {
-    jobUpdateBehaviour.completeJobUpdate(command);
+    final long jobKey = command.getKey();
+    jobUpdateBehaviour
+        .getJob(jobKey, command)
+        .ifRightOrLeft(
+            job -> {
+              final List<String> errors = new ArrayList<>();
+              jobUpdateBehaviour.updateJobRetries(jobKey, job, command).ifPresent(errors::add);
+              final long timeout = command.getValue().getTimeout();
+              // if no timeout is provided (the default value is -1L), no update
+              if (timeout > -1L) {
+                jobUpdateBehaviour.updateJobTimeout(jobKey, job, command).ifPresent(errors::add);
+              }
```

Comment:
🔧 Whatever the timeout update validation logic is, should it be the same for `UPDATE` and `UPDATE_TIMEOUT` commands? Currently, we have an additional validation step here in the `UPDATE` processor.

Reply:
The problem is that we are telling the users that:

`Providing any of those attributes with a null value or omitting it preserves the persisted attribute’s value.`

This is not possible if we don't add this additional validation

Reply:
From the REST controller, we send a `0` if nothing is provided by the user, right?

Reply:
Yes, so I need to change the validation 👍